Roles alter

mariacha opened the pull request:
Changes the names of the two provided roles to "Developer" for what used to be "administrator" and "content editor" for what used to be "site editor".

The motivation behind this is to reduce the number of users assigned to the administrator role.

Special attention needs to be paid to what a user with just the "content editor" permissions can do. Can they view published and unpublished content? Can they alter menus? What other things should a content-centric role be allowed to do?

This is not ready to be merged.

mariacha committed:
Removing give admins all roles.

mariacha committed:
Adding new permissions for admins and content editors.

mariacha commented:
Question -- what levels of permissions should the content editor have with panel pages? Create new? Edit existing? Panelizer?

jhedstrom commented on os_project/modules/features/os_base/os_base.install:
```diff
@@ -92,21 +92,4 @@ function os_base_install() {
-    user_role_save($admin_role);
-    user_role_grant_permissions($admin_role->rid, array_keys(module_invoke_all('permission')));
-    // Set this as the administrator role.
-    variable_set('user_admin_role', $admin_role->rid);
```
Doesn't this variable still need to be set?

mariacha committed:
Content editors need to be able to make menu items.

mariacha committed:
We're going to say only the developer can add page manager pages.
Since panel pages aren't enabled by default, no permissions for content editors here.

mariacha commented:
My proposed list of permissions is:

1) Add a new page
2) Add media
3) Approve pages made by others
4) Add and move blocks?
5) Use contextual links
6) Do with Documents, Audio, Video, Image
7) View site reports
8) View user pages
9) Create menu links

Developer should be able to:
1) Administer anything
2) Not make content
3) Use Page manager

Would it make sense to do behat tests for this?

mariacha committed:
Beginnings of behat tests.

jhedstrom commented on os_project/modules/features/os_base/os_base.info:
```diff
@@ -12,10 +12,14 @@ dependencies[] = filter
 dependencies[] = media
+dependencies[] = navbar
 dependencies[] = node
+dependencies[] = overlay
```
Let's kill this and the navbar dependency (most devs disable overlay, this would prevent them from doing so).

jhedstrom commented:
:+1: for the behat tests! I added a few code comments above.
